homework/index.js

```diff
@@ -30,17 +34,96 @@
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
+    });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
+  }
+
+  function renderContributions(repo, ul) {
+    fetchJSON(repo.contributors_url, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+      ul.innerHTML = '';
+      createAndAppend('h3', ul, { text: `Contributors:` });
+      contributors.forEach(contributor => {
+        const a = createAndAppend('a', ul, { href: contributor.html_url, target: '_blank' });
+        const li = createAndAppend('li', a);
+        const table = createAndAppend('table', li);
+        const tbody = createAndAppend('tbody', table);
+        const tr1 = createAndAppend('tr', tbody);
+        createAndAppend('img', tr1, { src: contributor.avatar_url });
+        createAndAppend('td', tr1, { text: contributor.login });
+        createAndAppend('td', tr1, { text: contributor.contributions });
+      });
+    });
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repo, index) => {
+        createAndAppend('option', select, { text: repo.name, value: index });
+      });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header);
+    const mainContainer = createAndAppend('div', root, { id: `main-container` });
+    const div1 = createAndAppend('div', mainContainer, { id: 'table-container' });
+    const div2 = createAndAppend('div', mainContainer, { id: 'list-container' });
+    const ul = createAndAppend('ul', div2, { id: 'list-contributions' });
+
     fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+
+      function setDefault() {
+        renderRepos(repositories[0], div1);
+        renderContributions(repositories[0], ul);
+      }
+      setDefault();
+
+      createOptionElements(repositories, select);
```

Comment:
You have a bug in your code. As you can see in the picture below, in the select box the `alumni` repo name is shown, whereas the details are shown for the `AngularJS` repo:

![image](https://user-images.githubusercontent.com/2788771/62125203-c8982f00-b2cc-11e9-99dc-fc56c3eca6ce.png)

Can you solve this bug?

Reply:
I think I fixed it now